cinema/views.py

```diff
@@ -0,0 +1,43 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from cinema.models import Movie
+from cinema.serializers import MovieSerializer
+
+
+@api_view(["GET", "POST"])
+def movie_list(request):
+    if request.method == "GET":
+        movies = Movie.objects.all()
+        serializer = MovieSerializer(movies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    if request.method == "POST":
+        serializer = MovieSerializer(data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_details(request, pk):
+    movie = get_object_or_404(Movie, pk=pk)
+
+    if request.method == "GET":
+        serializer = MovieSerializer(movie)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    if request.method == "PUT":
+        serializer = MovieSerializer(movie, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_204_NO_CONTENT)
```

Comment:
wrong status

Reply:
You don't need to reload the page after a successful PUT request. When you send a PUT request to update a resource, the server should update the resource and return a response that indicates the update was successful.
The status status.HTTP_200_OK generally indicates that the request was successful and the server is returning a response with requested information. However, in this specific case, since the request is an update operation and not a retrieval operation, it might be more appropriate to return a status.HTTP_204_NO_CONTENT response to indicate that the server has successfully processed the request, but there is no content to return. This indicates to the client that the update was successful and there is no need to refresh or reload the resource.

Reply:
if it's the answer from gpt then it's wrong)
in API as a rule, after you PUT or PATCH some object, the changed object is returned, so 204 is not good for sure

Reply:
<img width="900" alt="Screenshot 2023-04-05 at 11 53 05" src="https://user-images.githubusercontent.com/104159378/230031891-6eda36b3-bb45-441a-88f0-33d0bf8fe552.png">
<img width="906" alt="Screenshot 2023-04-05 at 11 52 53" src="https://user-images.githubusercontent.com/104159378/230031900-44cfa8a7-2a7b-472e-a6f7-24680429e821.png">
You can see what statuses DRF returns on put and patch

Reply:
Yup I even checked with most API, which were listed to me by GPT and every returned 200 instead of 204 declaimed by him... Well, at least now I know that xD